### PR TITLE
Switch to macos-12

### PR DESCRIPTION
**Pull request by cqc-alec**
Because it seems apple-clang 13.1 (macos-12) is not quite ABI-compatible with apple-clang 13.0 (macos-11), we have to build our own symengine package for this to work. I copied the recipe exactly from the conan-center, and uploaded the packages.

### CODE REVIEW

**Comment by cqc-melf** [comment]
The macos 11 check is still required but not running? I assume that is intended?

**Comment by cqc-alec** [comment]
> The macos 11 check is still required but not running? I assume that is intended?

Yes, I will change it after merging this.

File: recipes/symengine/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+sources:
+  "0.8.1":
+    url: "https://github.com/symengine/symengine/releases/download/v0.8.1/symengine-0.8.1.tar.gz"
+    sha256: "41eb6ae6901c09e53d7f61f0758f9201e81fc534bfeecd4b2bd4b4e6f6768693"
+  "0.9.0":
```
Comment (cqc-melf) [inline recipes/symengine/all/conandata.yml:5]: Why are we using two different versions?

File: .github/workflows/build-symengine.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+name: Build symengine
+
+on:
+  push:
+    branches:
+      - 'symengine-upload-new/**'
```
Comment (cqc-melf) [inline .github/workflows/build-symengine.yml:6]: do you want to have this on other branches, too?

File: recipes/symengine/all/conandata.yml (hunk 1 above)
Comment (cqc-alec) [reply recipes/symengine/all/conandata.yml:5]: We are only using 0.9.0 (as that is the version specified in the tket conanfile). We could delete the other option from here, but it does no harm (and is copied directly from the conan-center recipe). When new versions come out, we can just add them here and update tket retaining the ability to revert if there are problems.

File: .github/workflows/build-symengine.yml (hunk 2 above)
Comment (cqc-alec) [reply .github/workflows/build-symengine.yml:6]: No, the idea is that in order to upload a new symengine version (which won't happen very often), one just has to create a branch with a name matching that pattern, modify the symengine recipe and push it.

Comment (cqc-melf) [reply .github/workflows/build-symengine.yml:6]: Sounds like a good idea! Thank you for explaining this.

**Review by cqc-melf — DISMISSED** [review]

**Comment by cqc-alec** [comment]
Just merged from develop and bumped the tket version again.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**